service/server.py

```diff
@@ -185,50 +185,54 @@ def display_title(title_ref):
         abort(404)
 
 
-@app.route('/title-search/', methods=['GET', 'POST'])
+@app.route('/title-search', methods=['GET', 'POST'])
+@app.route('/title-search/<search_term>', methods=['GET', 'POST'])
 @login_required
-def find_titles():
-    # TODO: make this method use a WTF form, just like signin()
-    if request.method == "POST":
+def find_titles(search_term=''):
+    if request.method == 'POST':
         search_term = request.form['search_term'].strip()
-        LOGGER.info(
-            "SEARCH REGISTER: '{0}' was searched by {1}".format(
-                search_term,
-                current_user.get_id()))
-        # Determine search term type and preform search
-        title_number_regex = re.compile(TITLE_NUMBER_REGEX)
-        postcode_regex = re.compile(BASIC_POSTCODE_REGEX)
-        search_term = search_term.upper()
-        # If it matches the title number regex...
-        if title_number_regex.match(search_term):
-            title = get_register_title(search_term)
-            if title:
-                # If the title exists store it in the session
-                session['title'] = title
-                # Redirect to the display_title method to display the digital
-                # register
-                return redirect(url_for('display_title', title_ref=search_term))
-            else:
-                # If title not found display 'no title found' screen
-                return render_search_results([], search_term)
-        # If it matches the postcode regex ...
-        elif postcode_regex.match(search_term):
-            # Short term fix to enable user to search with postcode without spaces
-            postcode = sanitise_postcode(search_term)
-            postcode_search_results = get_register_titles_via_postcode(postcode)
-            return render_search_results(postcode_search_results, postcode)
-        elif search_term:
-            address_search_results = get_register_titles_via_address(search_term)
-            return render_search_results(address_search_results, search_term)
+        if search_term:
+            return redirect(url_for('find_titles', search_term=search_term))
         else:
+            # display the initial search page
+            return render_template('search.html', asset_path='/static/',
```

Comment:
How about redirecting here, as well? When the user refreshes the page, they will be asked for resubmission confirmation.